### PR TITLE
fix(flow): Call setFocus() on back button click

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** None

## Summary

- Depends on #7252
- When the back button is clicked, `setFocus()` is called and the newly active panel is focused.
- Adds test

### CODE REVIEW

File: packages/calcite-components/src/components/flow/flow.e2e.ts [hunk 1]
```diff
@@ -55,6 +56,24 @@ describe("calcite-flow", () => {
...
+    it("should call setFocus() on back button click", async () => {
+      const page = await newE2EPage();
+
+      await page.setContent(
+        html`<calcite-flow
+          ><calcite-flow-item id="one"></calcite-flow-item><calcite-flow-item id="two"></calcite-flow-item
+        ></calcite-flow>`
+      );
+
+      await page.$eval("#two", (elm: HTMLCalciteFlowItemElement) =>
```
Comment (jcfranco) [inline packages/calcite-components/src/components/flow/flow.e2e.ts:68]: In order to remove implementation details, can we click the back button vs firing the custom event?

File: packages/calcite-components/src/components/flow/flow.e2e.ts [hunk 2]
```diff
@@ -55,6 +56,24 @@ describe("calcite-flow", () => {
...
+        elm.dispatchEvent(new CustomEvent("calciteFlowItemBack"))
+      );
+      await page.waitForTimeout(0);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/flow/flow.e2e.ts:71]: Curious, do we still need the timeout delay after the `setFocus` changes? If so, maybe we could add a new util (separately) to ensure we wait consistently for `setFocus`-related tests.

**Review by jcfranco — APPROVED** [review]
Sweet, thanks for splitting this up! 🪵🪓

File: packages/calcite-components/src/components/flow/flow.e2e.ts (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/components/flow/flow.e2e.ts]: yup!

File: packages/calcite-components/src/components/flow/flow.e2e.ts (hunk 2 above)
Comment (driskull) [reply packages/calcite-components/src/components/flow/flow.e2e.ts]: Seems to not be necessary here. Ill remove.

**Pull request merged**